consensus/service/src/byzantine_ledger.rs

```diff
@@ -709,113 +707,115 @@ impl<
 
     fn attempt_complete_cur_slot(&mut self) {
         // See if we have externalized values for the current slot.
-        let ext_vals = self.scp.get_externalized_values(self.cur_slot);
-        if ext_vals.is_empty() {
-            return;
-        }
 
-        // Update pending value processing time metrics.
-        for ext_val in ext_vals.iter() {
-            if let Some(Some(timestamp)) = self.pending_values_map.get(ext_val) {
-                let duration = Instant::now().saturating_duration_since(*timestamp);
-                counters::PENDING_VALUE_PROCESSING_TIME.observe(duration.as_secs_f64());
-            }
-        }
-
-        // Maintain the invariant that pending_values only contains valid values that were not externalized.
-        self.pending_values
-            .retain(|tx_hash| !ext_vals.contains(tx_hash));
+        match self.scp.get_externalized_values(self.cur_slot) {
+            None => {}
+            Some(ext_vals) => {
+                // Update pending value processing time metrics.
+                for value in ext_vals.iter() {
+                    if let Some(Some(timestamp)) = self.pending_values_map.get(value) {
+                        let duration = Instant::now().saturating_duration_since(*timestamp);
+                        counters::PENDING_VALUE_PROCESSING_TIME.observe(duration.as_secs_f64());
+                    }
+                }
 
-        log::info!(
-            self.logger,
-            "Slot {} ended with {} externalized values and {} pending values.",
-            self.cur_slot,
-            ext_vals.len(),
-            self.pending_values.len(),
-        );
+                // Invariant: pending_values only contains valid values that were not externalized.
+                self.pending_values
+                    .retain(|tx_hash| !ext_vals.contains(tx_hash));
 
-        // Write to ledger.
-        {
-            let (block, block_contents, signature) = self
-                .tx_manager
-                .tx_hashes_to_block(&ext_vals)
-                .unwrap_or_else(|e| panic!("Failed to build block from {:?}: {:?}", ext_vals, e));
+                log::info!(
+                    self.logger,
+                    "Slot {} ended with {} externalized values and {} pending values.",
+                    self.cur_slot,
+                    ext_vals.len(),
+                    self.pending_values.len(),
+                );
 
-            log::info!(
-                self.logger,
-                "Appending block {} to ledger (sig: {}, tx_hashes: {:?}).",
-                block.index,
-                signature,
-                ext_vals,
-            );
+                // Write to ledger.
+                {
+                    let (block, block_contents, signature) = self
+                        .tx_manager
+                        .tx_hashes_to_block(&ext_vals)
+                        .unwrap_or_else(|e| {
```

Comment:
My understanding is that we're striving to avoid this pattern in favor of `expect("Failed to build block")`, wtih the error enum including the details in it's Display implementation. Then `expect()` will panic with the output "Failed to build block: {err}" @jcape